src/axom/sina/core/File.hpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef SINA_FILE_HPP
+#define SINA_FILE_HPP
+
+/*!
+ ******************************************************************************
+ *
+ * \file File.hpp
+ *
+ * \brief   Implementation file for Sina File class
+ *
+ ******************************************************************************
+ */
+
+#include <string>
+#include <vector>
+
+#include "conduit.hpp"
+
+namespace axom
+{
+namespace sina
+{
+/**
+ * \brief An object to help track locations of files in the file system
+ *
+ * A File tracks the location (URI) and mimetype of a file on the file system, plus any tags.
+ * In the Sina schema, a File always belongs to a Record or one of Record's inheriting types.
+ *
+ * Every File must have a URI, while mimetype and tags are optional.
+ *
+ * \code
+ *   axom::sina::File myFile{"/path/to/file.png"};
+ *   myFile.setMimeType("image/png");
+ *   axom::sina::File myOtherFile{"/path/to/other/file.txt"};
+ *   myOtherFile.setTags({"these","are","tags"});
+ *   myRecord->add(myFile);
+ *   myRecord->add(myOtherFile);
+ * \endcode
+ */
+class File
```

Comment:
Note: Axom's `core` component has a `Path` class that might be relevant.

See: https://github.com/LLNL/axom/blob/develop/src/axom/core/Path.hpp